standard/expressions.md

```diff
@@ -4594,6 +4622,7 @@ The `??` operator is called the null coalescing operator.
 null_coalescing_expression
     : conditional_or_expression
     | conditional_or_expression '??' null_coalescing_expression
+    | throw_expression
```

Comment:
I think this allows `throw throw throw throw...` Probably should prevent that!

Reply:
Surely though the grammar can prevent it directly without any semantic rules?

I think (i.e. I haven't tested it through ANTLR for any issues) the following does this:

```ANTLR
null_coalescing_expression
    : conditional_or_expression
    | conditional_or_expression '??' (null_coalescing_expression | throw_expression)
    ;
```

Reply:
You still need the semantic rule, otherwise you could write `F(throw null)`, which the prose is intended to forbid.

Given the prose, there is no need to make the grammar more complex.

Reply:
I'm afraid I'm going to disagree here. If the semantic rule is required for other contexts that a throw expression can occur then of course it must stay. However that in no way argues against the grammar being made more precise for *null_coalescing_expression* – the grammar is not really "more complex", adding a couple of parentheses doesn’t do that, but it is clearer.

Reply:
Given @jskeet’s recent comment here are two options to improve the grammar. First:

```ANTLR
null_coalescing_expression
    : conditional_or_expression
    | conditional_or_expression '??' (null_coalescing_expression | throw_expression)
    ;
```

This avoids allowing `throw throw throw...` in the *grammar* but does introduce an “RE” rule, if that is seen as too complex then:

```ANTLR
null_coalescing_expression
    : conditional_or_expression
    | conditional_or_expression '??' null_coalescing_expression
    | conditional_or_expression '??' throw_expression
    ;
```

can be used.

 I vote for the first myself.

Reply:
We would also have to modify the grammar for `?:` and every function member declaration.  I don't want to complexify the spec that way.

Reply:
The whole v7 grammar is going need to be checked, just as was done for v6, by "someone" ;-) to check all the additions do what was intended and don't alter existing parses unintentionally, that the semantic rules fit in, etc. Let's just leave the grammar here as is for now, I don't think anything that is wrong *per se* has been noted yet, its a debate over style. "Someone" will maybe get to the overall task before the EOY...